src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/Logviewer.java

```diff
@@ -170,6 +170,18 @@ public void scrollUpToCenter() {
     WebElement element = Helper.locateElementByCssSelector(
       Helper.getCssSelectorByDataTestId("log-viewer-content")
     );
+
+    Integer scrollHeight = Integer.valueOf(element.getDomProperty("scrollHeight"));
+    String heightInPixels = element.getCssValue("height");
+    Integer height = Integer.valueOf(heightInPixels.substring(0, heightInPixels.length() - 2));
+    Integer waitTime = 0;
+
+    //check if scrollbar is present
+    while (scrollHeight < height + 50 && waitTime < 180) {
+      Helper.waitSeconds(5);
+      waitTime += 5;
+      scrollHeight = Integer.valueOf(element.getDomProperty("scrollHeight"));
+    }
```

Comment:
Preferably use ExpectedConditions as demonstrated in this stackoverflow answer https://stackoverflow.com/a/43116592

Reply:
Done